src/main/java/hudson/plugins/sonar/client/WsClient.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Jenkins Plugin for SonarQube, open source software quality management tool.
+ * mailto:contact AT sonarsource DOT com
+ *
+ * Jenkins Plugin for SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Jenkins Plugin for SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package hudson.plugins.sonar.client;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+
+public class WsClient {
+  private static final String API_MEASURES = "/api/measures/component";
+  private static final String API_PROJECT_STATUS = "/api/qualitygates/project_status";
+  private static final String API_VERSION = "/api/server/version";
+  private static final String API_PROJECT_NAME = "/api/projects/index";
+
+  private final HttpClient client;
+  private final String serverUrl;
+  private final String username;
+  private final String password;
+  private final String projectKey;
+
+  public WsClient(HttpClient client, String serverUrl, String projectKey, String username, String password) {
+    this.client = client;
+    this.serverUrl = serverUrl;
+    this.projectKey = projectKey;
+    this.username = username;
+    this.password = password;
+  }
+
+  public ProjectQualityGate getQualityGate52() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00112)